### PR TITLE
Add favorites/bookmarks functionality to videos with dedicated screen

### LINKED ISSUE #300: Create the SavedScreen

### PR DESCRIPTION
**Pull request by Fayroth**
This PR implements a comprehensive favorites system for videos in EduVerse, allowing users to bookmark content and access it through a dedicated saved posts screen.

## Key Changes
- Add bookmark button to video player interface for quick saving
- Implement bookmark/favorite button in publication detail dialog
- Create new SavedScreen accessible from settings menu to view all bookmarked content
- Reuse existing favorites repository methods and components
- Add state management for favorite status
- Implement extensive UI tests for the new SavedScreen

## Implementation Details
- Positioned bookmark button consistently in video UI action buttons layout
- Added visual feedback for favorite status (yellow when favorited)
- Implemented proper state management to track favorite status
- Created dedicated SavedScreen with grid view of bookmarked posts
- Added empty state handling when no posts are saved
- Reused existing PublicationItem and PublicationDetailDialog components
- Comprehensive test coverage following existing patterns

## Testing
- Added complete test suite for SavedScreen
- Tests cover: screen loading, navigation, empty states, grid display
- Verified video indicator visibility
- Tested publication dialog interactions
- Created necessary fake implementations for testing

### Video Player with Bookmark Button
Video Player

### Settings Navigation
Detail View

### Saved Posts Screen
| Empty State | Grid View |
|------------|-----------|
|Empty State|Grid View|

### Video feed
Settings
## Related Issues
Closes #300

### CODE REVIEW

File: app/src/main/java/com/github/se/eduverse/ui/setting/SavedScreen.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  val favoritePublications by viewModel.favoritePublications.collectAsState()
+  var selectedPublication by remember {
+    mutableStateOf<com.github.se.eduverse.model.Publication?>(null)
+  }
+
+  LaunchedEffect(userId) { viewModel.loadFavoritePublications(userId) }
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
```
Comment (MathieuxC) [inline app/src/main/java/com/github/se/eduverse/ui/setting/SavedScreen.kt:37]: You should use TopNavigationBar, or at least follow the same color pattern

**Review by MathieuxC — CHANGES_REQUESTED** [review]
lgtm, you should just change the topBar of SavedScreen to align with the rest of the app

File: app/src/main/java/com/github/se/eduverse/ui/setting/SavedScreen.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                  onClick = { navigationActions.goBack() },
+                  modifier = Modifier.testTag("back_button")) {
+                    Icon(Icons.Default.ArrowBack, contentDescription = "Back")
+                  }
+            },
+            colors =
+                TopAppBarDefaults.smallTopAppBarColors(
+                    containerColor = MaterialTheme.colorScheme.primary,
+                    titleContentColor = MaterialTheme.colorScheme.onPrimary,
+                    navigationIconContentColor = MaterialTheme.colorScheme.onPrimary))
```
Comment (abj10) [inline app/src/main/java/com/github/se/eduverse/ui/setting/SavedScreen.kt:50]: **Refactor Top Bar:**
Use our generalized `TopNavigationBar` for `SavedScreen`, or implement at least the gradient color scheme if you think that the `TopNavigationBar` is not suited to your changes so that we align with other screens design.

File: app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt [hunk 3]
```diff
@@ -293,6 +294,41 @@ fun VideoScreen(
...
+                                    onClick = {
+                                      profileViewModel.toggleFavorite(currentUserId, publication.id)
+                                      isFavorited.value = !isFavorited.value
+                                      Log.d(
+                                          "BOOKMARK",
+                                          "Bookmark button clicked for publication: ${publication.id}")
+                                    },
+                                    modifier =
+                                        Modifier.align(Alignment.CenterEnd)
+                                            .offset(y = 256.dp) // Positioned below the share button
```
Comment (abj10) [inline app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt:318]: It's preferred that offset would not be hardcoded

**Review by abj10 — CHANGES_REQUESTED** [review]
# Review for PR : Add favorites/bookmarks functionality to videos with a dedicated screen

## Summary
This PR introduces an interesting bookmarking feature to the application, allowing users to save favorite videos with a new "Saved Posts" screen effectively. It includes good but improvable updates to the UI, navigation logic, and repository methods, along with extensive test coverage.

## Strengths

1. **Feature Integration:**
   - Implemented a dedicated `SavedScreen` to display favorited posts.
   - Bookmarking actions are well-integrated across the application, with clear UI feedback (e.g., `BookmarkAdd` icon with color changes for state differentiation).

2. **Navigation Updates:**
   - Added seamless navigation to the new `SavedScreen` using `Screen.SAVED`.
   - Navigation actions are clearly defined and tested for back navigation (`goBack`).

3. **Test Coverage:**
   - Comprehensive tests for the new screen (`SavedScreenTest`) cover edge cases like empty states, interaction with publications, and dialog visibility.
   - Modified `ProfileScreenTest` ensures compatibility with new features.

4. **User Feedback:**
   - Clear UI updates for bookmarking actions with tags like `BookmarkedIcon` and `UnbookmarkedIcon`.
   - Proper use of Material Design components ensures consistency with the existing app style.

## Weaknesses

1. **Top Bar Color Consistency:**
   - The `SavedScreen` `TopAppBar` does not use `TopNavigationBar` or follow the same gradient styling pattern seen in other screens like `Videos`.

2. **Code Duplication:**
   - Repeated logic for checking and toggling the bookmark state in `VideoScreen` and `PublicationDetailDialog`. Consider creating a helper method to handle this.

3. **Hardcoded UI Values:**
   - The `offset(y = 256.dp)` in the `VideoScreen` for the bookmark button position might lead to layout inconsistencies on different screen sizes.

## Changes requested

1. **Refactor Top Bar:**
   - Use `TopNavigationBar` for `SavedScreen` or implement the gradient color scheme to align with other screens.

## Suggestions for Improvement

1. **Consolidate Bookmark Logic:**
   - Create a utility function in the ViewModel or repository to handle bookmark toggling and state retrieval.

2. **Replace Hardcoded Dimensions:**
   - Replace `offset(y = 256.dp)` and similar values with dynamic spacing based on layout constraints or themes.

3. **Enhance Repository Mocking:**
   - While the `FakeProfileRepository` handles testing needs, the stubs (`TODO`) for some methods are incomplete. Fully implement mock behaviors for consistency.

## Conclusion

This PR adds a valuable bookmarking feature, improving user experience and engagement. While the implementation is robust. I requested changes on the 'TopAppBar', and I suggest minor refinements on code reuse, which would make it more polished.

**Comment by abj10** [comment]
Great, everything LGTM now. Good Job.

**Review by abj10 — APPROVED** [review]
Great, everything LGTM now. Good Job.

File: app/src/main/java/com/github/se/eduverse/ui/setting/SavedScreen.kt (hunk 1 above)
Comment (Fayroth) [reply app/src/main/java/com/github/se/eduverse/ui/setting/SavedScreen.kt]: yes you're absolutely right ive made the changes

**Review by MathieuxC — APPROVED** [review]
